### PR TITLE
[mxnet][tensorflow][tests] useIMDSv2_Code_Update and fix OSS license git clone issues

### PR DESCRIPTION
**Pull request by jeet4320**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/container_tests/bin/testMXNetHVD [hunk 1]
```diff
@@ -1,6 +1,11 @@
 #!/bin/bash
 
-INSTANCE_TYPE=`curl http://169.254.169.254/latest/meta-data/instance-type`
+INSTANCE_TYPE=$1
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testMXNetHVD:3]: Can we completely remove the `INSTANCE_TYPE` from the script. We can handle the skipping the test based on the instance type using pytest fixtures.

File: test/dlc_tests/container_tests/bin/testMXNetHVD [hunk 2]
```diff
@@ -44,7 +49,6 @@ echo "Train a CNN model using Horovod with MXNet on MNIST data. You can follow p
 if [ ${RETURN_VAL} -eq 0 ]; then
   GPU_AMOUNT=`nvidia-smi -L | wc -l`
   if [ $GPU_AMOUNT -ge 8 ]; then GPU_AMOUNT=8; fi
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testMXNetHVD:51]: Can we confirm removing this does not hamper the test execution?

File: test/dlc_tests/container_tests/bin/testTF2HVD [hunk 3]
```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
-INSTANCE_TYPE=`curl http://169.254.169.254/latest/meta-data/instance-type`
+INSTANCE_TYPE=$1
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testTF2HVD:2]: Same here, can be entirely removed

File: test/dlc_tests/container_tests/bin/testTF1HVDHelper [hunk 4]
```diff
@@ -32,7 +38,6 @@ echo "Train a resnet model using Horovod with Tensorflow on synthetic data. You
 if [ ${RETURN_VAL} -eq 0 ]; then
   GPU_AMOUNT=`nvidia-smi -L | wc -l`
   if [ $GPU_AMOUNT -ge 8 ]; then GPU_AMOUNT=8; fi
```
Comment (tejaschumbalkar) [inline test/dlc_tests/container_tests/bin/testTF1HVDHelper:40]: Can we confirm removing this does not hamper the test execution?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 5]
```diff
@@ -469,14 +469,17 @@ def test_oss_compliance(image):
...
 
-            try:
-                if not os.path.isdir(local_file_path):
-                    context.run(f"git clone {url.rstrip()} {local_file_path}")
-                    context.run(f"tar -czvf {local_file_path}.tar.gz {local_file_path}")
-            except Exception as e:
-                LOGGER.error(f"Unable to clone git repo. Error: {e}")
-                raise
-
+            for i in range(3):
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_pre_release.py:472]: Is this change related to IMDSv2 update?

File: test/dlc_tests/container_tests/bin/testMXNetHVD (hunk 2 above)
Comment (jeet4320) [reply test/dlc_tests/container_tests/bin/testMXNetHVD:51]: you mean removing GPU_AMOUNT? why do remove this?

File: test/dlc_tests/sanity/test_pre_release.py (hunk 5 above)
Comment (jeet4320) [reply test/dlc_tests/sanity/test_pre_release.py:472]: Nope, change PR title to address this change

**Review by tejaschumbalkar — COMMENTED** [review]
nit changes to be addressed in subsequent PR. Current changes LGTM.

**Review by tejaschumbalkar — APPROVED** [review]
